### PR TITLE
Add default:river_water_* to minimal game

### PR DESCRIPTION
**Pull request by Rogier-5**
When starting, minetest complains that mapgen_river_water_source
is not defined.

Alternatively, a river water node could be added. This was the
easiest option. If a river water node is preferred, I'll do that
instead.

### CODE REVIEW

**Comment by paramat** [comment]
Normally it would be best to just alias normal water, however any mapgen with sloping rivers (mgvalleys) needs a non-renewable water node with range = 2, so adding river water to minimal is actually essential here.

**Comment by Rogier-5** [comment]
> adding river water to minimal is actually essential here

 OK. Done.

**Comment by paramat** [comment]
Will review properly after release.

File: games/minimal/mods/default/init.lua [hunk 1]
```diff
@@ -1005,6 +1005,60 @@ minetest.register_node("default:water_source", {
...
+	buildable_to = true,
+	is_ground_content = false,
+	drowning = 1,
+	liquidtype = "flowing",
+	liquid_alternative_flowing = "default:river_water_flowing",
+	liquid_alternative_source = "default:river_water_source",
+	liquid_viscosity = WATER_VISC,
+	liquid_renewable = false,
+	liquid_range = 2,
+	post_effect_color = {a=64, r=100, g=100, b=200},
```
Comment (paramat) [inline games/minimal/mods/default/init.lua]: Post effect colour is different for river water.

File: games/minimal/mods/default/init.lua [hunk 2]
```diff
@@ -1005,6 +1005,60 @@ minetest.register_node("default:water_source", {
...
+	buildable_to = true,
+	is_ground_content = false,
+	drowning = 1,
+	liquidtype = "source",
+	liquid_alternative_flowing = "default:river_water_flowing",
+	liquid_alternative_source = "default:river_water_source",
+	liquid_viscosity = WATER_VISC,
+	liquid_renewable = false,
+	liquid_range = 2,
+	post_effect_color = {a=64, r=100, g=100, b=200},
```
Comment (paramat) [inline games/minimal/mods/default/init.lua]: As above.

File: games/minimal/mods/default/init.lua [hunk 3]
```diff
@@ -1005,6 +1005,60 @@ minetest.register_node("default:water_source", {
...
 })
 
+minetest.register_node("default:river_water_flowing", {
+	description = "River water (flowing)",
+	inventory_image = minetest.inventorycube("default_river_water.png"),
+	drawtype = "flowingliquid",
+	tiles ={"default_river_water.png"},
+	special_tiles = {
+		{name="default_river_water.png", backface_culling=false},
+		{name="default_river_water.png", backface_culling=true},
```
Comment (paramat) [inline games/minimal/mods/default/init.lua]: 4 lines above, spaces around '='

File: games/minimal/mods/default/init.lua [hunk 4]
```diff
@@ -1005,6 +1005,60 @@ minetest.register_node("default:water_source", {
...
+})
+
+minetest.register_node("default:river_water_source", {
+	description = "Water",
+	inventory_image = minetest.inventorycube("default_river_water.png"),
+	drawtype = "liquid",
+	tiles ={"default_river_water.png"},
+	special_tiles = {
+		-- New-style river_water source material (mostly unused)
+		{name="default_river_water.png", backface_culling=false},
```
Comment (paramat) [inline games/minimal/mods/default/init.lua]: 4 lines above, spaces around '='.

File: games/minimal/mods/default/init.lua [hunk 5]
```diff
@@ -1005,6 +1005,60 @@ minetest.register_node("default:water_source", {
...
+	is_ground_content = false,
+	drowning = 1,
+	liquidtype = "flowing",
+	liquid_alternative_flowing = "default:river_water_flowing",
+	liquid_alternative_source = "default:river_water_source",
+	liquid_viscosity = WATER_VISC,
+	liquid_renewable = false,
+	liquid_range = 2,
+	post_effect_color = {a=64, r=100, g=100, b=200},
+	groups = {water=3, liquid=3},
```
Comment (paramat) [inline games/minimal/mods/default/init.lua]: Spaces around '='.

File: games/minimal/mods/default/init.lua [hunk 6]
```diff
@@ -1005,6 +1005,60 @@ minetest.register_node("default:water_source", {
...
+	is_ground_content = false,
+	drowning = 1,
+	liquidtype = "source",
+	liquid_alternative_flowing = "default:river_water_flowing",
+	liquid_alternative_source = "default:river_water_source",
+	liquid_viscosity = WATER_VISC,
+	liquid_renewable = false,
+	liquid_range = 2,
+	post_effect_color = {a=64, r=100, g=100, b=200},
+	groups = {river_water=3, liquid=3},
```
Comment (paramat) [inline games/minimal/mods/default/init.lua]: Spaces around '='.

File: games/minimal/mods/default/init.lua [hunk 7]
```diff
@@ -1005,6 +1005,60 @@ minetest.register_node("default:water_source", {
 	groups = {water=3, liquid=3},
 })
 
+minetest.register_node("default:river_water_flowing", {
+	description = "River water (flowing)",
+	inventory_image = minetest.inventorycube("default_river_water.png"),
```
Comment (paramat) [inline games/minimal/mods/default/init.lua]: We don't use inventory cube anymore.

File: games/minimal/mods/default/init.lua [hunk 8]
```diff
@@ -1005,6 +1005,60 @@ minetest.register_node("default:water_source", {
...
+	liquid_viscosity = WATER_VISC,
+	liquid_renewable = false,
+	liquid_range = 2,
+	post_effect_color = {a=64, r=100, g=100, b=200},
+	groups = {water=3, liquid=3},
+})
+
+minetest.register_node("default:river_water_source", {
+	description = "Water",
+	inventory_image = minetest.inventorycube("default_river_water.png"),
```
Comment (paramat) [inline games/minimal/mods/default/init.lua]: We don't use inventory cube anymore.

Comment (C1ffisme) [inline games/minimal/mods/default/init.lua]: Maybe I'm an idiot, but if you look up a couple of lines you'll see that minimal still uses inventory cube for normal water.

Comment (paramat) [inline games/minimal/mods/default/init.lua]: The engine doesn't use inventory cube anymore as we have a new way to render the inventory items, so use of it should ideally be removed throughout minimal.

Comment (C1ffisme) [inline games/minimal/mods/default/init.lua]: Ah, okay. Thanks for the clarification.

**Comment by paramat** [comment]
@Rogier-5 this is needed please fix.

**Comment by paramat** [comment]
No response, closing.
I might do this properly later.

**Comment by paramat** [comment]
Replaced by #5809

**Pull request closed without merging**